lib/modules.nix

```diff
@@ -141,7 +144,11 @@ rec {
         config = mkMerge [ (removeAttrs m ["_file" "key" "disabledModules" "require" "imports"]) metaSet ];
       };
 
-  applyIfFunction = key: f: args@{ config, options, lib, ... }: if isFunction f then
+
+  applyIfFunction = key: f: args@{ config, options, lib, ... }:
+    # We only want to apply module arguments if we're actually dealing with a
+    # module function, an opaque wrapper isn't one
+    if isFunction f && ! isOpaqueWrapper f then
```

Comment:
This right here is a bit hacky. It's needed because the module system assumes that if you have a function as a module, it must be a NixOS module function. This worked fine with `imports = [ (mkMerge [ ... ]) ];` previously, because `mkMerge` didn't return a function but an attrset. But with this change it does. So we need to prevent it trying to call the opaque wrapper function (which is really just an implementation detail) with NixOS arguments.

I'll argue that the module system allowing both attribute sets and functions as modules on a guessing basis is hacky already, so this is just a counter-hack!